specs/wp-log-in-out-spec.js

```diff
@@ -136,6 +203,140 @@ test.describe( `[${host}] Authentication: (${screenSize}) @parallel @jetpack`, f
 			} );
 		} );
 	}
+
+	if ( dataHelper.hasAccountWithFeatures( '+passwordless +2fa-sms' ) ) {
+		test.describe( 'Can Log in on a passwordless account', function() {
+			test.describe( 'Can request a magic link email by entering the email of an account which does not have a password defined', function() {
+				let magicLoginLink, loginFlow, magicLinkEmail, emailClient;
+				test.before( function () {
+					loginFlow = new LoginFlow( driver, [ '+passwordless',  '+2fa-sms' ] );
+					emailClient = new EmailClient( get( loginFlow.account, 'mailosaur.inboxId' ) );
+					return loginFlow.login();
+				} );
+
+				test.it( 'Can find the magic link in the email received', function() {
+					return emailClient.pollEmailsByRecipient( loginFlow.account.email ).then( function( emails ) {
+						magicLinkEmail = emails.find( email => email.subject.indexOf( 'WordPress.com' ) > -1 );
+						assert( magicLinkEmail !== undefined, 'Could not find the magic login email' );
+						magicLoginLink = magicLinkEmail.html.links[0].href;
+						assert( magicLoginLink !== undefined, 'Could not locate the magic login link in the email' );
+						return true;
+					} );
+				} );
+
+				test.describe( 'Can use the magic link and the code received via sms to log in', function() {
+					let magicLoginPage, twoFALoginPage, twoFACode;
+					test.before( function( done ) {
+						driver.get( magicLoginLink );
+						magicLoginPage = new MagicLoginPage( driver );
+						// make sure we listen for SMS before we trigger any
+						const xmppClient = listenForSMS( loginFlow.account );
+						xmppClient.once( 'online', () => {
+							// send sms now!
+							magicLoginPage.finishLogin();
+							twoFALoginPage = new LoginPage( driver );
+							twoFALoginPage.use2FAMethod( 'sms' );
```

Comment:
Because we chose an account that has 2FA method set to sms,
the screen that will be displayed after successful magic link usage
will be the right one for the 2FA method...

So do we actually need to call it and if we do, do we need to tell it what method to use?

Reply:
We might want to pick an account that has multiple types of 2fa enabled, also it's not reliable which 2fa screen is displayed, if you had a failure once you will be displayed with a different screen on your next attempt (see my other comment)